### ISSUE TITLE
Windows 10: stuck in "Building" step

### ISSUE THREAD

**Issue by mantielero** [nimterop/nimterop#138]
I am trying to install `nimterop` in a coorporate Windows 10 machine. I did:
```
choosenim 0.20.2
```
and everything went fine.

And now:
```
c:\folders>nimble --verbose install nimterop@#head
    Reading official package list
Downloading https://github.com/genotrance/nimterop using git
  Verifying dependencies for nimterop@#head
    Reading official package list
   Checking for regex@>= 0.10.0
      Info: Dependency on regex@>= 0.10.0 already satisfied
  Verifying dependencies for regex@0.11.2
    Reading official package list
   Checking for unicodedb@>= 0.7.0
      Info: Dependency on unicodedb@>= 0.7.0 already satisfied
  Verifying dependencies for unicodedb@0.7.1
    Reading official package list
   Checking for unicodeplus@>= 0.5.0
      Info: Dependency on unicodeplus@>= 0.5.0 already satisfied
  Verifying dependencies for unicodeplus@0.5.0
    Reading official package list
   Checking for unicodedb@>= 0.7
      Info: Dependency on unicodedb@>= 0.7 already satisfied
  Verifying dependencies for unicodedb@0.7.1
    Reading official package list
   Checking for cligen@>= 0.9.17
      Info: Dependency on cligen@>= 0.9.17 already satisfied
  Verifying dependencies for cligen@0.9.31
 Installing nimterop@#head
   Building nimterop/nimterop/toast.exe using c backend
```
It gets stuck there.

**Comment by mantielero** [comment]
After checking the task manager, it shows that it gets stuck with C/C++ compiler rather than the git process.
![task_manager](https://user-images.githubusercontent.com/1116064/62023618-f6e51400-b1d1-11e9-8e00-13359a581c4a.png)

**Comment by mantielero** [comment]
When the proccess `cc1` is killed:
```
Tip: 11 messages have been suppressed, use --verbose to show them.
Error: Build failed for package: nimterop
 ... Details:
 ... Execution failed with exit code 1
 ... Command: "C:\Users\theuser\.nimble\bin\nim.exe" c --noBabelPath -d:release --path:"C:\Users\theuser\.nimble\pkgs\regex-0.11.2"  --path:"C:\Users\theuser\.nimble\pkgs\unicodedb-0.7.1"  --path:"C:\Users\theuser\.nimble\pkgs\unicodeplus-0.5.0"  --path:"C:\Users\theuser\.nimble\pkgs\unicodedb-0.7.1"  --path:"C:\Users\theuser\.nimble\pkgs\cligen-0.9.31"  -o:"C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop\toast.exe" "C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop\toast.nim"
 ... Output: Hint: used config file 'C:\Users\theuser\.choosenim\toolchains\nim-0.20.2\config\nim.cfg' [Conf]
 ... Hint: used config file 'C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\config.nims' [Conf]
 ... Hint: system [Processing]
 ... Hint: widestrs [Processing]
 ... Hint: io [Processing]
 ... Hint: toast [Processing]
 ... Hint: os [Processing]
 ... Hint: strutils [Processing]
 ... Hint: parseutils [Processing]
 ... Hint: math [Processing]
 ... Hint: bitops [Processing]
 ... Hint: macros [Processing]
 ... Hint: algorithm [Processing]
 ... Hint: unicode [Processing]
 ... Hint: pathnorm [Processing]
 ... Hint: osseps [Processing]
 ... Hint: winlean [Processing]
 ... Hint: dynlib [Processing]
 ... Hint: times [Processing]
 ... Hint: options [Processing]
 ... Hint: typetraits [Processing]
 ... Hint: time_t [Processing]
 ... Hint: strformat [Processing]
 ... Hint: api [Processing]
 ... Hint: setup [Processing]
 ... Hint: git [Processing]
 ... Hint: osproc [Processing]
 ... Hint: strtabs [Processing]
 ... Hint: hashes [Processing]
 ... Hint: streams [Processing]
 ... Hint: cpuinfo [Processing]
 ... Hint: regex [Processing]
 ... Hint: sets [Processing]
 ... Hint: tables [Processing]
 ... Hint: properties [Processing]
 ... Hint: properties_data [Processing]
 ... Hint: types [Processing]
 ... Hint: types_data [Processing]
 ... Hint: unicodeplus [Processing]
 ... Hint: paths [Processing]
 ... Hint: compat [Processing]
 ... Hint: types [Processing]
 ... # Setting up Git repo: https://github.com/tree-sitter/tree-sitter
 ... # Checking out 0.15.5
 ... # Setting up Git repo: https://github.com/JuliaStrings/utf8proc
 ... # Pulling repository
 ... Hint: c [Processing]
 ... # Setting up Git repo: https://github.com/tree-sitter/tree-sitter-c
 ... # Checking out v0.15.0
 ... Hint: cpp [Processing]
 ... # Setting up Git repo: https://github.com/tree-sitter/tree-sitter-cpp
 ... # Checking out v0.15.0
 ... Hint: ast [Processing]
 ... Hint: sequtils [Processing]
 ... Hint: getters [Processing]
 ... Hint: globals [Processing]
 ... Hint: plugin [Processing]
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop\globals.nim(20, 5) Warning: Deprecated since v0.20, use 'toHashSet'; toSet is deprecated [Deprecated]
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop\globals.nim(28, 5) Warning: Deprecated since v0.20, use 'toHashSet'; toSet is deprecated [Deprecated]
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop\getters.nim(27, 28) Warning: Deprecated since v0.20, use 'toHashSet'; toSet is deprecated [Deprecated]
 ... Hint: grammar [Processing]
 ... Hint: lisp [Processing]
 ... Hint: cligen [Processing]
 ... Hint: parseopt3 [Processing]
 ... Hint: critbits [Processing]
 ... Hint: argcvt [Processing]
 ... Hint: textUt [Processing]
 ... Hint: terminal [Processing]
 ... Hint: colors [Processing]
 ... Hint: wordwrap [Processing]
 ... Hint: sysUt [Processing]
 ... Hint: macUt [Processing]
 ... CC: lib
 ... CC: parser
 ... CC: parser
 ... CC: scanner
 ... CC: stdlib_assertions.nim
 ... CC: stdlib_dollars.nim
 ... CC: stdlib_widestrs.nim
 ... CC: stdlib_io.nim
 ... CC: stdlib_system.nim
 ... CC: stdlib_parseutils.nim
 ... CC: stdlib_macros.nim
 ... CC: stdlib_math.nim
 ... CC: stdlib_algorithm.nim
 ... CC: stdlib_unicode.nim
 ... CC: stdlib_strutils.nim
 ... CC: stdlib_pathnorm.nim
 ... CC: stdlib_dynlib.nim
 ... CC: stdlib_winlean.nim
 ... CC: stdlib_times.nim
 ... CC: stdlib_os.nim
 ... CC: stdlib_strformat.nim
 ... CC: stdlib_hashes.nim
 ... CC: stdlib_streams.nim
 ... CC: stdlib_osproc.nim
 ... CC: stdlib_sets.nim
 ... CC: stdlib_tables.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/unicodedb-0.7.1/unicodedb/properties_data.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/unicodedb-0.7.1/unicodedb/properties.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/unicodedb-0.7.1/unicodedb/types_data.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/unicodedb-0.7.1/unicodedb/types.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/unicodeplus-0.5.0/unicodeplus.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/regex-0.11.2/regex.nim
 ... CC: git.nim
 ... CC: stdlib_sequtils.nim
 ... CC: plugin.nim
 ... CC: globals.nim
 ... CC: getters.nim
 ... CC: lisp.nim
 ... CC: grammar.nim
 ... CC: ast.nim
 ... CC: stdlib_critbits.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/cligen-0.9.31/cligen/parseopt3.nim
 ... CC: stdlib_terminal.nim
 ... CC: stdlib_wordwrap.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/cligen-0.9.31/cligen/textUt.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/cligen-0.9.31/cligen/argcvt.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/cligen-0.9.31/cligen/sysUt.nim
 ... CC: ../../../../Users/theuser/.nimble/pkgs/cligen-0.9.31/cligen.nim
 ... CC: toast.nim
 ... Error: execution of an external compiler program 'gcc.exe -c  -w -mno-ms-bitfields -DWIN32_LEAN_AND_MEAN -DUTF8PROC_STATIC -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib/include -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib/src -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib/../../utf8proc -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter_c\src -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter_cpp\src -O3 -fno-strict-aliasing  -IC:\Users\theuser\.choosenim\toolchains\nim-0.20.2\lib -IC:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\nimterop -o C:\Users\theuser\nimcache\toast_r\lib.c.o C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c' failed with exit code: 1
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:10:0:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./get_changed_ranges.c: In function 'ts_range_array_intersects':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./get_changed_ranges.c:28:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (unsigned i = start_index; i < self->size; i++) {
 ...    ^
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./get_changed_ranges.c:28:3: note: use option -std=c99 or -std=gnu99 to compile your code
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./get_changed_ranges.c: In function 'iterator_descend':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./get_changed_ranges.c:205:5: error: 'for' loop initial declarations are only allowed in C99 mode
 ...      for (uint32_t i = 0, n = ts_subtree_child_count(*entry.subtree); i < n; i++) {
 ...      ^
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:11:0:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./language.c: In function 'ts_language_symbol_for_name':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./language.c:54:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (TSSymbol i = 0; i < count; i++) {
 ...    ^
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./language.c: In function 'ts_language_field_id_for_name':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./language.c:96:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (TSSymbol i = 1; i < count + 1; i++) {
 ...    ^
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:12:0:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./lexer.c: In function 'ts_lexer_goto':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./lexer.c:199:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (unsigned i = 0; i < self->included_range_count; i++) {
 ...    ^
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:13:0:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./node.c: In function 'ts_subtree_has_trailing_empty_descendant':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./node.c:177:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (unsigned i = ts_subtree_child_count(self) - 1; i + 1 > 0; i--) {
 ...    ^
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./parser.c:16:0,
 ...                  from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:14:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\././reduce_action.h: In function 'ts_reduce_action_set_add':
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\././reduce_action.h:22:3: error: 'for' loop initial declarations are only allowed in C99 mode
 ...    for (uint32_t i = 0; i < self->size; i++) {
 ...    ^
 ... In file included from C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\lib.c:14:0:
 ... C:\Temp\nimble_10188\githubcom_genotrancenimterop_#head\build\inc\treesitter\lib\src\./parser.c: In function 'ts_parser__log':
```

**Comment by genotrance** [comment]
What version of gcc do you have installed? I have `{.passC: "-std=c11".}` but your gcc isn't using it.

**Comment by mantielero** [comment]
I had:
```
C:>gcc --version
gcc (GCC) 4.8.3
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Solved with version 5.1.0 from [tdm-gcc](http://tdm-gcc.tdragon.net/download).
